### ISSUE TITLE
YAML file IAM policies erroneously trigger "E2507 IAM Policy Documents needs to be JSON"

### ISSUE THREAD

**Issue by alexjurkiewicz** [awslabs/cfn-python-lint#105]
A resource like this:

```yaml
Resources:
  MyUser:
    Type: AWS::IAM::User
    Properties:
      Policies:
        - PolicyName: !Sub ${AWS::StackName}-policy
          PolicyDocument:
            - Version: '2012-10-17'
              Statement:
                Action:
                  - s3:GetObject
                Effect: Allow
                Resource:
                  - arn:aws:s3:::*
```

Will be created just fine by the Amazon API, but cfn-lint reports:

```
E2507 IAM Policy Documents needs to be JSON
test.yaml:7:11

E3002 Property should be of type Json not List at Resources/MyUser/Properties/Policies/0/PolicyDocument
test.yaml:7:11
```

**Comment by kddejong** [comment]
thanks @alexjurkiewicz I'll get on this one.

**Comment by kddejong** [comment]
@alexjurkiewicz 

```yaml
CFNUser:
  Type: AWS::IAM::User
  Properties:
    Policies:
    - PolicyName: !Sub ${AWS::StackName}-policy
      PolicyDocument:
        - Version: '2012-10-17'
          Statement:
            Action:
              - s3:GetObject
            Effect: Allow
            Resource:
              - arn:aws:s3:::*
```

This gives me an error of `Value of property PolicyDocument must be an object`

**Comment by kddejong** [comment]
Sorry ended that a little earlier.  I've been trying to replicate your success and I'm having troubles.  

In fact cut and pasted your example and got the same error  `Value of property PolicyDocument must be an object`

**Comment by kddejong** [comment]
@alexjurkiewicz so I can completely replicate what you are doing what region are deploying into?

**Comment by cmmeyer** [comment]
Hi @alexjurkiewicz just checking in if you can provide details to help us replicate?  Is this still an issue?
